src/server/scripts/Northrend/AzjolNerub/Ahnkahet/boss_jedoga_shadowseeker.cpp

```diff
@@ -15,589 +15,487 @@
+    SPELL_CYCLONE_STRIKE          = 56855,
+    SPELL_SACRIFICE_BEAM          = 56150,
+
+    //Jedoga Controller
+    SPELL_BEAM_VISUAL_JEDOGA      = 56312,
+    SPELL_SACRIFICE_VISUAL        = 56133,
+
+    //Twilight Volunteer
+    SPELL_SPHERE_VISUAL_VOLUNTEER = 56102,
+    SPELL_PILLAR_OF_LIGHTNING     = 56868
 };
 
-const Position JedogaPosition[2] =
+enum JedogaEvents
 {
-    {372.330994f, -705.278015f, -0.624178f,  5.427970f},
-    {372.330994f, -705.278015f, -16.179716f, 5.427970f}
+    EVENT_INTRO_SAY = 1,
+    EVENT_START_FIGHT_1,
+    EVENT_START_FIGHT_2,
+    EVENT_CYCLONE_STRIKE,
+    EVENT_LIGHTNING_BOLT,
+    EVENT_THUNDERSHOCK,
+    EVENT_START_PHASE_TWO,
+    EVENT_FLY_DELAY,
+    EVENT_END_PHASE_TWO,
+    EVENT_CHOOSE_VOLUNTEER,
+    EVENT_SUMMON_VOLUNTEER
 };
 
-enum Misc
+enum JedogaPhases
 {
-    ACTION_INITIAND_KILLED                      = 1,
-    DATA_VOLUNTEER_WORK                         = 2
+    PHASE_INTRO = 0,
+    PHASE_ONE,
+    PHASE_TWO,
+    PHASE_THREE
 };
 
-class boss_jedoga_shadowseeker : public CreatureScript
+enum JedogaPoints
 {
-public:
-    boss_jedoga_shadowseeker() : CreatureScript("boss_jedoga_shadowseeker") { }
-
-    struct boss_jedoga_shadowseekerAI : public ScriptedAI
-    {
-        boss_jedoga_shadowseekerAI(Creature* creature) : ScriptedAI(creature)
-        {
-            Initialize();
-            instance = creature->GetInstanceScript();
-            bFirstTime = true;
-            bPreDone = false;
-        }
-
-        void Initialize()
-        {
-            uiOpFerTimer = urand(15 * IN_MILLISECONDS, 20 * IN_MILLISECONDS);
-
-            uiCycloneTimer = 3 * IN_MILLISECONDS;
-            uiBoltTimer = 7 * IN_MILLISECONDS;
-            uiThunderTimer = 12 * IN_MILLISECONDS;
-
-            bOpFerok = false;
-            bOpFerokFail = false;
-            bOnGround = false;
-            bCanDown = false;
-            volunteerWork = true;
-        }
+    POINT_INITIAL_POSITION = 0,
+    POINT_SACRIFICE,
+    POINT_GROUND,
+    POINT_PHASE_TWO,
+    POINT_PHASE_TWO_FLY
+};
 
-        InstanceScript* instance;
+Position const SacrificePosition = { 376.5385f, -707.3567f, -16.14124f };
+Position const JedogaGroundPosition = { 371.6281f, -704.4836f, -16.17967f };
+Position const JedogaFlyPosition = { 371.627f, -704.4217f, -6.707521f };
+Position const JedogaControllerPositions[3] =
+{
+    { 402.7893f, -748.5251f, 29.39399f, 2.024582f },
+    { 420.1999f, -727.0132f, 28.88036f, 2.042035f },
+    { 375.4977f, -707.3635f, -16.0964f, 2.426008f }
+};
 
-        uint32 uiOpFerTimer;
-        uint32 uiCycloneTimer;
-        uint32 uiBoltTimer;
-        uint32 uiThunderTimer;
+typedef std::pair<Position, Position> VolunteerPositionPair;
+std::vector<VolunteerPositionPair> VolunteerSpotPositions =
+{
+    { { 400.7701f, -784.8928f, -31.60143f }, { 365.9514f, -719.1235f, -16.17974f } },
+    { { 397.3595f, -788.5157f, -31.59679f }, { 359.7433f, -715.017f,  -16.17974f } },
```

Comment:
Can I keep this part as it is or should I be using something like bellow?
```cpp
std::make_pair(Position(400.7701f, -784.8928f, -31.60143f), Position(365.9514f, -719.1235f, -16.17974f)),

Reply:
no need to